registration/include/pcl/registration/registration.h

```diff
@@ -368,9 +368,9 @@ namespace pcl
        * \param[in] visualizerCallback reference of the user callback function
        */
       template<typename FunctionSignature> inline bool
-      registerVisualizationCallback (boost::function<FunctionSignature> &visualizerCallback)
+      registerVisualizationCallback (std::function<FunctionSignature> &visualizerCallback)
```

Comment:
I'm unsure on what to do here. For this to work, it needs to have the same type as `update_visualizer_`. According to line 575
```cpp
      std::function<void(const pcl::PointCloud<PointSource> &cloud_src,
                           const std::vector<int> &indices_src,
                           const pcl::PointCloud<PointTarget> &cloud_tgt,
                           const std::vector<int> &indices_tgt)>
```

The first thing that comes to mind is expose the callback signature type in the public API  and apply it to the method signature.

Reply:
The original intent behind having `FunctionSignature` as a template parameter is not clear to me. The author should have typedefed the callback type and used it, would spare us the trouble. But what's done is done.

So now I see two ways forward. First is as you suggested; it's simple but may break some user code. Arguably, this is a debugging API, so we are not breaking anything critical. Second idea is to change the template argument from `FunctionSignature` to `FunctionT` and spell out the signature explicitly:

```cpp
template<template<typename> typename FunctionT> inline bool
registerVisualizationCallback (FunctionT<void(const pcl::PointCloud<PointSource> &cloud_src,
                                              const std::vector<int> &indices_src,
                                              const pcl::PointCloud<PointTarget> &cloud_tgt,
                                              const std::vector<int> &indices_tgt)> &visualizerCallback)
```

At the same time we may still introduce a typedef and switch to typedef-only signarute in next releases.

Reply:
I will go with the non-breakage option and add the typedef.